Pod/Classes/OAStackViewDistributionStrategy.m

```diff
@@ -203,8 +212,107 @@ - (void)addProportionalityConstraintsBetween:(UIView *)view otherView:(UIView *)
+      return isInitialConstraint ? NSLayoutAttributeTop : NSLayoutAttributeBottom;
+  }
+}
+
+- (NSMutableArray *)equalSpacingLayoutGuides
+{
+  if (!_equalSpacingLayoutGuides) {
+    _equalSpacingLayoutGuides = [NSMutableArray array];
+  }
+
+  return _equalSpacingLayoutGuides;
+}
+
+- (NSString *)symbolicSpacingRelation
+{
+  return @">=";
+}
+
+- (void)alignMiddleView:(UIView *)view afterView:(UIView *)previousView
+{
+  [super alignMiddleView:view afterView:previousView];
+
+  _OALayoutGuide *guide = [_OALayoutGuide new];
+  [self.equalSpacingLayoutGuides addObject:guide];
+  [self.stackView addSubview:guide];
+
+  NSMutableArray *newConstraints = [NSMutableArray array];
+
+  UILayoutConstraintAxis axis = self.stackView.axis;
+
+  NSLayoutConstraint *firstEdgeConstraint =
+  [NSLayoutConstraint constraintWithItem:guide
+                               attribute:[self spanningAttributeForAxis:axis
+                                                          isInitialEdge:YES]
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:previousView
+                               attribute:[self spanningAttributeForAxis:axis
+                                                          isInitialEdge:NO]
+                              multiplier:1
+                                constant:0];
+
+  NSLayoutConstraint *secondEdgeConstraint =
+  [NSLayoutConstraint constraintWithItem:view
+                               attribute:[self spanningAttributeForAxis:axis
+                                                          isInitialEdge:YES]
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:guide
+                               attribute:[self spanningAttributeForAxis:axis
+                                                          isInitialEdge:NO]
+                              multiplier:1
+                                constant:0];
+
+  [newConstraints addObjectsFromArray:@[firstEdgeConstraint, secondEdgeConstraint]];
+
+  id firstGuide = self.equalSpacingLayoutGuides.firstObject;
+  if (firstGuide) {
+    NSLayoutConstraint *equalWidth =
+    [NSLayoutConstraint constraintWithItem:firstGuide
+                                 attribute:[self equalityAxis]
+                                 relatedBy:NSLayoutRelationEqual
+                                    toItem:guide
+                                 attribute:[self equalityAxis]
+                                multiplier:1
+                                  constant:0];
+
+    equalWidth.identifier = @"OA-fill-equally";
+
+    [newConstraints addObject:equalWidth];
+  }
+
+  [self.constraints addObjectsFromArray:newConstraints];
+  [self.stackView addConstraints:newConstraints];
+}
+
+- (void)removeAddedConstraints
+{
+  [self.stackView removeConstraints:self.constraints];
+  [self.constraints removeAllObjects];
+
+  for (UIView *v in self.equalSpacingLayoutGuides) {
```

Comment:
What about using `[self.equalSpacingLayoutGuides makeObjectsPerformSelector:@selector(removeFromSuperview)];` instead of the for loop here?